timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -107,7 +129,7 @@ def __init__(self, data: Any, static_features: Optional[pd.DataFrame] = None, *a
                 self._validate_multi_index_data_frame(data)
             else:
                 data = self._construct_pandas_frame_from_data_frame(data)
-        elif isinstance(data, str):
+        elif isinstance(data, (str, Path)):
```

Comment:
why not do `static_features: Any` and also accept a path there while we're at it?